### PR TITLE
13295 explorer: use anchor links for subsections

### LINKED ISSUE #13295: explorer: Use anchor links for subsections
#### Problem
Explorer doesn't provide anchor links for the various sections displayed. For example, on the transaction page, it would be nice if you could link someone directly to the "Account Inputs" section.

#### Proposed Solution
- Add anchor links to all major sections
- Update current url w/ anchor hash as user scrolls

### PR DESCRIPTION
**Pull request by KitanGarcia**
#### Problem    
Explorer subsections did not have anchors attached to them for easy navigation

#### Summary of Changes    
* Added anchors linking to the subsections/headers throughout Explorer

Fixes #13295

### CODE REVIEW

**Comment by oJshua** [comment]
@KitanGarcia this isn't working as intended, for example when I visit [http://localhost:3000/address/SysvarC1ock11111111111111111111111111111111#history](url) the browser should scroll down to the Transaction History subsection.

**Comment by KitanGarcia** [comment]
> @KitanGarcia this isn't working as intended, for example when I visit [http://localhost:3000/address/SysvarC1ock11111111111111111111111111111111#history](url) the browser should scroll down to the Transaction History subsection.

Thanks for reviewing and getting back to me! I have an idea on how to do and I'll hope to get to it soon. Apologies for such a late reply!

**Comment by KitanGarcia** [comment]
> @KitanGarcia this isn't working as intended, for example when I visit [http://localhost:3000/address/SysvarC1ock11111111111111111111111111111111#history](url) the browser should scroll down to the Transaction History subsection.

Hi @oJshua , I have a couple ideas on how to do this. Since the reason it isn't working as expected is due to different components rendering at different times, I see 2 approaches:
1) Using Refs and scrolling to them in a useEffect after X seconds.    
```
  useEffect(() => {
    if (hashRef && hashRef.current) {
      setTimeout(function () {
        hashRef!.current!.scrollIntoView();
      }, 3000);
    }
  }, [hashRef]);

  return (
    <div className="card">
      <div className="card-header">
        <h4 id="txStats" className="card-header-title">
          <a href="#txStats" ref={hashRef}>
            Live Transaction Stats
          </a>
        </h4>
      </div>
      <TpsCardBody />
    </div>
  );
```
This works (at least for most), but is not so nice, since it would have to wait and it may not work for those with poor connection speeds.    

2) Creating skeletons for components that would be the same height and width as the component they represent so that the anchor element would not move out of focus when all components have finished rendering. This is probably cleaner, but would probably take a long time to build out.

Would you have a preference between these methods, or an alternative?

**Comment by KitanGarcia** [comment]
@oJshua Just thought I'd ask once more if you'd have a preference or alternative solution regarding my above post: https://github.com/solana-labs/solana/pull/23151#issuecomment-1083976236

**Comment by jstarry** [comment]
@steveluscher is this PR obsoleted by your recent changes?

File: explorer/src/components/SupplyCard.tsx [hunk 1]
```diff
@@ -66,7 +66,9 @@ const renderHeader = () => {
     <div className="card-header">
       <div className="row align-items-center">
         <div className="col">
-          <h4 className="card-header-title">Supply Overview</h4>
+          <h4 id="overview" className="card-header-title">
+            <a href="#overview">Supply Overview</a>
```
Comment (steveluscher) [inline explorer/src/components/SupplyCard.tsx:70]: I made a new bulletproof system for scrolling to anchors! Check it out in
#24861.

**Review by steveluscher — CHANGES_REQUESTED** [review]
If you want to update this PR to treat these interesting sections as scroll anchors using the new `useScrollAnchor()` infra, please do and I'll accept it.

File: explorer/src/components/SupplyCard.tsx (hunk 1 above)
Comment (steveluscher) [reply explorer/src/components/SupplyCard.tsx:70]: Essentially it's this:

```
import { useScrollAnchor } from "providers/scroll-anchor";
/* ... */
const scrollAnchorRef = useScrollAnchor('overview');
/* ... */
<h4 className="card-header-title" ref={scrollAnchorRef}>
  Supply Overview
</h4>
```

**Comment by steveluscher** [comment]
Feel free to reopen this if you'd like to take a second crack at it with the new API!

**Pull request closed without merging**